File: domain/src/main/java/com/ahn/domain/currency/repository/CurrencySelectionRepository.kt
package com.ahn.domain.currency.repository

import com.ahn.domain.currency.model.CalculatorCurrencySelection
import com.ahn.domain.currency.model.ExchangeCurrencySelection

interface CurrencySelectionRepository {
    suspend fun getCalculatorSelection(): CalculatorCurrencySelection
    suspend fun saveCalculatorMainCurrencyCode(code: String)
    suspend fun saveCalculatorSubCurrencyCode(code: String)
    suspend fun saveCalculatorSelection(mainCode: String, subCode: String)

    suspend fun getExchangeSelection(): ExchangeCurrencySelection
    suspend fun saveExchangeFromCurrencyCode(code: String)
    suspend fun saveExchangeToCurrencyCode(code: String)
    suspend fun saveExchangeSelection(fromCode: String, toCode: String)
}

File: ios/Packages/Domain/Sources/Domain/Currency/Repository/CurrencySelectionRepository.swift
public protocol CurrencySelectionRepository: Sendable {
    func calculatorSelection() async throws -> CalculatorCurrencySelection

    func saveCalculatorMainCurrencyCode(_ code: String) async throws

    func saveCalculatorSubCurrencyCode(_ code: String) async throws

    func saveCalculatorSelection(mainCode: String, subCode: String) async throws

    func exchangeSelection() async throws -> ExchangeCurrencySelection

    func saveExchangeFromCurrencyCode(_ code: String) async throws

    func saveExchangeToCurrencyCode(_ code: String) async throws

    func saveExchangeSelection(fromCode: String, toCode: String) async throws
}
